modules/misc/gtk.nix

```diff
@@ -166,14 +166,35 @@ in {
     home.packages = optionalPackage cfg.font ++ optionalPackage cfg.theme
       ++ optionalPackage cfg.iconTheme;
 
-    home.file.${cfg2.configLocation}.text =
-      concatStringsSep "\n" (mapAttrsToList formatGtk2Option ini) + "\n"
-      + cfg2.extraConfig;
+    home.file.${cfg2.configLocation} = {
+      text = concatStringsSep "\n" (mapAttrsToList formatGtk2Option ini) + "\n"
+        + cfg2.extraConfig;
+      onChange = mkIf (cfg.theme != null)
+        (let xsettingsd = "${pkgs.xsettingsd}/bin/xsettingsd";
+        in ''
+          echo "Reloading gtk2"
+          $DRY_RUN_CMD ${xsettingsd} -c \
+          <( echo "Net/ThemeName \"${cfg.theme.name}\"" ) & \
+          process=$!
+          sleep 0.3 && kill $process || true
+          unset process
```

Comment:
```suggestion
          $DRY_RUN_CMD timeout 0.3 ${xsettingsd} -c <(echo 'Net/ThemeName "${cfg.theme.name}"') &
```

Reply:
I actually ran into problems with a short timeout and had to increase it. 2s should be fine, and it doesn't hurt anyway.

Reply:
Thanks for the feedback and heads up about #2239. Since it'll probably be merged before this, i'm gonna go ahead and remove `$DRY_RUN_CMD` like you did with the other programs.